kubernetes-client/src/main/java/io/fabric8/kubernetes/client/impl/KubernetesClientImpl.java

```diff
@@ -730,7 +736,10 @@ public void visitResources(ApiVisitor visitor) {
         .withVersions(new GroupVersionForDiscoveryBuilder().withGroupVersion("v1").build()).build()))) {
       return; // user terminated
     }
-    visitGroups(visitor, getApiGroups().getGroups());
+    APIGroupList apiGroups = getApiGroups();
+    if (apiGroups != null) {
```

Comment:
How come this may return null?

Reply:
The underlying rest call is allowed to return null on a 404.  However that isn't expected here.  It could be accounted for with an assertion instead.